tools/toolchains/arm.py

```diff
@@ -284,8 +285,9 @@ def link(self, output, objects, libraries, lib_dirs, scatter_file):
         if lib_dirs:
             args.extend(["--userlibpath", ",".join(lib_dirs)])
         if scatter_file:
-            new_scatter = self.correct_scatter_shebang(scatter_file)
-            args.extend(["--scatter", new_scatter])
+            scatter_name = split(scatter_file)[-1]
```

Comment:
I don't think this is quite right, shouldn't this be the path to the scatter_file relative to the project root @theotherjimmy ?

Reply:
Yes, this should be:
```suggestion
            scatter_name = dirname(scatter_file)
```

Reply:
I don't think dirname is correct either, right? That wouldn't include the actual file name.
Wouldn't it be closer to `relpath`?

Reply:
suggested changes by Jimmy have been made